data/simpy.gram

```diff
@@ -365,11 +373,37 @@ yield_expr[expr_ty]:
     | 'yield' 'from' a=expression { _Py_YieldFrom(a, EXTRA) }
     | 'yield' a=[expressions] { _Py_Yield(a, EXTRA) }
 
-arguments: expression for_if_clauses | args [',']
-args: '*' expression [',' args] | kwargs | posarg [',' args]  # Weird to make it work
-kwargs: ','.kwarg+
-posarg: named_expression | '*' expression
-kwarg: NAME '=' expression | '*' expression | '**' expression
+class_arguments[expr_ty]: a=args [','] { a }
+arguments[expr_ty]:
+    | a=expression b=for_if_clauses { _Py_Call(CONSTRUCTOR(p),
+                                               singleton_seq(p, _Py_GeneratorExp(a, b, EXTRA)),
+                                               NULL,
+                                               EXTRA) }
+    | a=args [','] { a }
+args[expr_ty]:
+    | a=starred_expression b=[',' c=args { c }] {
+        _Py_Call(CONSTRUCTOR(p),
+                 (b) ? seq_insert_in_front(p, a, ((expr_ty) b)->v.Call.args) : singleton_seq(p, a),
+                 (b) ? ((expr_ty) b)->v.Call.keywords : NULL,
+                 EXTRA) }
+    | a=kwargs { _Py_Call(CONSTRUCTOR(p),
+                          seq_extract_starred_exprs(p, a),
+                          seq_delete_starred_exprs(p, a),
+                          EXTRA) }
+    | a=posarg b=[',' c=args { c }] {
```

Comment:
Looks like we can get rid of `posarg` and instead write `a=named_expression` here -- the case where `posarg` matches `starred_expression` is already handled by the first alternative.

(Things are still weird -- one confusing thing is that `starred_expression` *must* start with `*`, while `named_expression` may or may not contain a `:=` operator. But I don't think we can simplify further, since the resolution of "ambiguities" is quite subtle for this rule.)